plugins/scaffolder/src/components/TaskPage/TaskPage.tsx

```diff
@@ -0,0 +1,336 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Page, Header, Lifecycle, Content } from '@backstage/core';
+import React, { useState, useEffect, memo, useMemo } from 'react';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+import Stepper from '@material-ui/core/Stepper';
+import Step from '@material-ui/core/Step';
+import StepLabel from '@material-ui/core/StepLabel';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import { generatePath, useParams } from 'react-router';
+import { useTaskEventStream } from '../hooks/useEventStream';
+import LazyLog from 'react-lazylog/build/LazyLog';
+import { Link } from 'react-router-dom';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Paper,
+  StepButton,
+  StepIconProps,
+} from '@material-ui/core';
+import { Status } from '../../types';
+import { DateTime, Interval } from 'luxon';
+import { useInterval } from 'react-use';
+import clsx from 'clsx';
+import Check from '@material-ui/icons/Check';
+import Cancel from '@material-ui/icons/Cancel';
+import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
+import { entityRoute } from '@backstage/plugin-catalog-react';
+import { parseEntityName } from '@backstage/catalog-model';
+import { TaskNotFound } from '../TaskNotFound/TaskNotFound';
+
+// typings are wrong for this library, so fallback to not parsing types.
+const humanizeDuration = require('humanize-duration');
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: '100%',
+    },
+    button: {
+      marginTop: theme.spacing(1),
+      marginRight: theme.spacing(1),
+    },
+    actionsContainer: {
+      marginBottom: theme.spacing(2),
+    },
+    resetContainer: {
+      padding: theme.spacing(3),
+    },
+    labelWrapper: {
+      display: 'flex',
+      flex: 1,
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+    },
+    stepWrapper: {
+      width: '100%',
+    },
+  }),
+);
+
+type TaskStep = {
+  id: string;
+  name: string;
+  status: Status;
+  startedAt?: string;
+  endedAt?: string;
+};
+
+const StepTimeTicker = ({ step }: { step: TaskStep }) => {
+  const [time, setTime] = useState('');
+
+  useInterval(() => {
+    if (!step.startedAt) {
+      setTime('');
+      return;
+    }
+
+    const end = step.endedAt
+      ? DateTime.fromISO(step.endedAt)
+      : DateTime.local();
+
+    const startedAt = DateTime.fromISO(step.startedAt);
+    const formatted = Interval.fromDateTimes(startedAt, end)
+      .toDuration()
+      .valueOf();
+
+    setTime(humanizeDuration(formatted, { round: true }));
+  }, 1000);
+
+  return <Typography variant="caption">{time}</Typography>;
+};
+
+const useStepIconStyles = makeStyles({
+  root: {
+    color: '#eaeaf0',
```

Comment:
does it work well with both dark and light mode? should try to avoid hard coded colors